### ISSUE TITLE
Select Options Fieldtype : default behaviour when field is required

### ISSUE THREAD

**Issue by er314** [processwire/processwire-issues#349]
I guess the observed behaviour is intended, but it's very surprising to me : 

When a Select Options field is "required", I would expect (I would want) that the user MUST position a value for this field, otherwise PW MUST refuse to save the page.

But the observed behaviour is that PW does position value id 1 by default.
And I can't seem to find a way to prevent this : in field settings, tried pre-selecting option blank to no avail ; tried setting default value = 0 to no avail.

Of course the user MAY change this default value. But this is misleading, and it's not "my intended behaviour".
In other words, when a Text field is "required", the user MUST position a value for this field ; PW does not position a default value "hello world" without my consent. Right ? :-)

PW 3.0.71
cheers

**Comment by Toutouwai** [comment]
I just encountered this problem. To restate the issue...

If "Select (Single value)" is chosen as the input type for a Select Options field, and the field is set to "required", the first option in the select input becomes pre-selected in Page Edit even if the "What options do you want pre-selected? (if any)" setting is none.

So despite the description of that setting being "This field also serves as a preview of your selected input type (InputfieldSelect) and options" it is in fact not an accurate preview.

Setting:
![2017-12-07_151903](https://user-images.githubusercontent.com/1538852/33695410-169dc172-db62-11e7-833a-2f831d26a553.png)

Page Edit:
![2017-12-07_151939](https://user-images.githubusercontent.com/1538852/33695418-1ea66d74-db62-11e7-8457-e2b0c587cc40.png)

If "Radios (Single value)" is chosen as the input type then no option is pre-selected (i.e. this input type works as expected).

**Comment by Toutouwai** [comment]
As @adrianbj [pointed out](https://processwire.com/talk/topic/10384-select-options-fieldtype-fieldtypeoptions/?do=findComment&comment=159614) in the forums, this is actually an issue with InputfieldSelect rather than FieldtypeOptions.

I think [this line](https://github.com/processwire/processwire/blob/6241fd67300c456c225cb336a3de27acab167b2d/wire/modules/Inputfield/InputfieldSelect.module#L263) could be amended to:

```php
if($allowBlank && !$this->attr('multiple') && !$hasBlankOption) {
```

No need that I can see to disallow the blank option if the inputfield is required.

**Comment by adrianbj** [comment]
Not a solution for the PW admin, but in case someone is mostly concerned about getting this to work on the frontend via the form API (which is my current need), this hack works:
```
$f->options = array('' => 'Choose One') + $f->options;
```
With this you can set required to true and get JS validation to work and still not have an initial valid option selected.

That said, I think the solution suggested by @Toutouwai sounds great!

**Comment by er314** [comment]
As per latest blog post : Tested in v3.0.93 -> this is not fixed.
And @Toutouwai solution seems to work for me.

**Comment by ryancramerdesign** [comment]
While this isn't specifically a bug, I agree that the behavior suggested here seems better, so have updated it to have that behavior when InputfieldSelect is used with an Options field. But since the InputfieldSelect behavior was intended, I'm going to leave the behavior as-is outside of use with an options field. After the next master, I'll switch the dev branch to update the behavior for other uses of InputfieldSelect, as I think that would need more testing just to make sure there aren't other side effects. Thanks.
